### PR TITLE
feat(taiko-client-rs): expose canShutdown on whitelist preconf /status

### PR DESCRIPTION
**Pull request by davidtaikocha**
## Summary

Mirrors [#21648](https://github.com/taikoxyz/taiko-mono/pull/21648) for the Rust whitelist preconfirmation driver. Adds a `canShutdown` boolean to `GET /status` so that Kubernetes `lifecycle.preStop` hooks can avoid recycling preconfer pods mid-window.

The Rust whitelist driver does not track lookahead operators, so the heuristic is simpler than the Go counterpart: `canShutdown` is `true` iff no `POST /preconfBlocks` request has been received within the last `1.5 × 8 slots × 12 s = 144 s`. A pod that has never received a request reports `true` (uninitialized = safe).

## Changes

- `WhitelistApiService` tracks `last_preconf_request_at: Mutex<Option<Instant>>`, stamped at the top of `build_preconf_block` so that even rejected requests block shutdown.
- `WhitelistStatus` and the REST `ApiStatus` gain `can_shutdown` (serialized as `canShutdown`).
- `/`, `/healthz`, `/status` are now JWT-skipped via a router split (public group + protected group); `/preconfBlocks` and `/ws` remain JWT-protected. Required because busybox `wget` (the k8s probe environment) can't forge HMAC-SHA256 bearer tokens.
- New unit tests: `can_shutdown_for` boundary cases; HTTP tests for the `canShutdown` JSON field, JWT-skip on probe paths, and JWT enforcement on protected paths.

## Test plan

- [x] `just fmt` — clean
- [x] `just clippy` — clean (strict gate, includes `-D missing_docs -D clippy::missing_docs_in_private_items`)
- [x] `cargo test -p whitelist-preconfirmation-driver --lib` — 89 passed, 0 failed
- [ ] Full `just test` (Dockerized e2e) — runs in CI
- [ ] CI on this PR

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: packages/taiko-client-rs/crates/whitelist-preconfirmation-driver/src/api/server/router.rs [hunk 1]
```diff
@@ -21,31 +21,42 @@ use super::{
...
+        protected_router = protected_router.route("/ws", get(handle_websocket_upgrade));
     }
 
-    router = router
+    let protected_router =
+        protected_router.layer(middleware::from_fn_with_state(state.clone(), auth_middleware));
+
+    public_router
+        .merge(protected_router)
         .fallback(handle_not_found)
```
[the remark this answers is not in the document]
Comment (davidtaikocha) [reply packages/taiko-client-rs/crates/whitelist-preconfirmation-driver/src/api/server/router.rs]: Valid issue. Confirmed the prior code applied `auth_middleware` after `.fallback(handle_not_found)` was set on the same router, which (per axum semantics) wrapped the fallback too — so `GET /unknown` with JWT configured returned 401. The router split moved the fallback to the outer post-merge router, dropping that protection.

Fixed in 23863f36e by placing `fallback(handle_not_found)` inside the protected sub-router so the auth layer wraps it. Two new tests pin the behavior:
- `unknown_path_requires_jwt_when_secret_configured` → 401
- `unknown_path_returns_not_found_when_no_jwt_secret` → 404

**Review by YoGhurt111 — APPROVED** [review]
